helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -1285,4 +1266,60 @@ public void handleSessionEstablishmentError(Throwable error) throws Exception {
   public Long getSessionStartTime() {
     return _sessionStartTime;
   }
+
+  private RealmAwareZkClient createSingleRealmZkClient() {
```

Comment:
- Rename the method to `prepareAndGetZkClient()`

- Javadoc:

Prepares connection config and client config based on the internal parameters given to HelixManager in order to create a ZkClient instance to use. Note that a shared ZkClient instance will be created if connecting as an ADMINISTRATOR to minimize the cost of creating ZkConnections.

Reply:
- `prepareAndGet` sounds like a method does 2 things. I prefer to name a method which does one thing. `get` is better than `prepareAndGet` as there is no need to care about "prepare" since the final purpose is "get". There is already one method `createClient()` so I prefer to keep the name.
- It is a private method and the logic is pretty straightforward. That's why I didn't consider adding comments. Anyway, since you suggested here, added.